Util/model_parser_cxx/model_parser.H

```diff
@@ -0,0 +1,308 @@
+#ifndef MODEL_PARSER_H
+#define MODEL_PARSER_H
+
+#include <iostream>
+#include <fstream>
+#include <vector>
+#include <algorithm>
+#include <network.H>
+#include <model_parser_data.H>
+
+using namespace amrex;
+
+///
+/// read in an initial model and return arrays with the model data.
+/// take care to match the species available in the model file to
+/// those defined by the network
+///
+/// the model file is assumed to be of the follow form:
+/// # npts = 896
+/// # num of variables = 6
+/// # density
+/// # temperature
+/// # pressure
+/// # carbon-12
+/// # oxygen-16
+/// # magnesium-24
+/// 195312.5000  5437711139.  8805500.952   .4695704813E+28  0.3  0.7  0
+/// 585937.5000  5410152416.  8816689.836  0.4663923963E+28  0.3  0.7  0
+///
+/// we read in the number of variables and their order and use this to
+/// map them into the model_state array.  We ignore anything other than
+/// density, temperature, pressure and composition.
+///
+/// composition is assumed to be in terms of mass fractions
+
+// remove whitespace -- from stackoverflow
+
+
+inline std::string& ltrim(std::string& s)
```

Comment:
Can we please put these functions in a class or namespace?

Reply:
done.